src/components/settings/SettingsLMSTab/LMSConfigs/BlackboardConfig.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Form, useToggle } from '@edx/paragon';
+import { buttonBool, handleErrors } from '../LMSConfigPage';
+import LmsApiService from '../../../../data/services/LmsApiService';
+import { snakeCaseDict } from '../../../../utils';
+import ConfigError from '../ConfigError';
+import ConfigModal from '../ConfigModal';
+
+import { SUCCESS_LABEL } from '../../data/constants';
+
+const BlackboardConfig = ({ id, onClick }) => {
+  const [clientId, setClientId] = React.useState('');
+  const [clientSecret, setClientSecret] = React.useState('');
+  const [blackboardBaseUrl, setBlackboardBaseUrl] = React.useState('');
+  const [errorIsOpen, openError, closeError] = useToggle(false);
+  const [modalIsOpen, openModal, closeModal] = useToggle(false);
+
+  const config = {
+    clientId,
+    clientSecret,
+    blackboardBaseUrl,
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const transformedConfig = snakeCaseDict(config);
+    // this will need to change based on save draft/submit
+    transformedConfig.active = false;
+    transformedConfig.enterprise_customer = id;
+    let err;
+    try {
+      await LmsApiService.postNewBlackboardConfig(transformedConfig);
+      err = undefined;
```

Comment:
is the setting to undefined needed? I would have thought that the original `let` assignment would have made it undefined.

Reply:
(this would also apply to all the other channel specific components)